Sources/Drain.swift

```diff
@@ -51,22 +49,15 @@ public final class Drain: DataRepresentable, Stream {
     }
 
     public func receive() throws -> Data {
-        let data = self.data
         close()
-        return data
+        return buffer
     }
 
     public func send(data: Data) throws {
-        enum Error: ErrorProtocol {
-            case sendUnsupported
-        }
-        throw Error.sendUnsupported
+        buffer += data.bytes
     }
-
+    
     public func flush() throws {
-        enum Error: ErrorProtocol {
-            case flushUnsupported
-        }
-        throw Error.flushUnsupported
+        buffer = []
     }
```

Comment:
don't you think it's better to use `nil` when we want to imply empty data? if not we should discuss this a bit more and maybe remove `NilLiteralConvertible` and use `[]` or `Data()`. which do you think conveys better the idea of an empty collection? as I'm typing I think `[]` is more appropriate haha.